src/lib/datepicker/calendar.ts

```diff
@@ -208,29 +230,37 @@ export class MatCalendar<D> implements AfterContentInit, OnDestroy, OnChanges {
     this._userSelection.emit();
   }
 
+  /** Handles month selection in the multi-year view. */
+  _yearSelected(year: D): void {
+    this._activeDate = year;
+    this._currentView = 'year';
+  }
+
   /** Handles month selection in the year view. */
   _monthSelected(month: D): void {
     this._activeDate = month;
-    this._monthView = true;
+    this._currentView = 'month';
   }
 
   /** Handles user clicks on the period label. */
   _currentPeriodClicked(): void {
-    this._monthView = !this._monthView;
+    this._currentView = this._currentView == 'month' ? 'multi-year' : 'month';
   }
 
   /** Handles user clicks on the previous button. */
   _previousClicked(): void {
-    this._activeDate = this._monthView ?
+    this._activeDate = this._currentView == 'month' ?
```

Comment:
Make a `_isMonthView()` function instead of repeating the comparison?

Reply:
Do we gain anything from the method? seems equally readable to what I'm doing here